client/state/signup/progress/reducer.js

```diff
@@ -74,6 +75,16 @@ function saveStep( state, { step } ) {
 	return addStep( state, { ...step, status: 'in-progress' } );
 }
 
+function setResumeAfterLogin( state, { resumeStep } ) {
+	debug( `Setting resume after login for step ${ resumeStep.stepName }` );
+	return map( state, function( step ) {
```

Comment:
I wonder, would something like this do the job? Or do we only want to set the `resumeStep` if it does already exist in the array? If that's not an issue we can avoid having to iterate over the collection.

```js
function setResumeAfterLogin( state, { resumeStep } ) {
  debug( `Setting resume after login for step ${ resumeStep.stepName }` );

  return {
    ...state,
    [ resumeStep.stepName ]: { 
      ...resumeStep, 
      status: 'in-progress' 
    },
  };
};
```

Reply:
`state` is an array so I think we unfortunately have to iterate through.

I see that other reducers in the file call `updateStep` to share this logic which might be an option here too. Its behaviour is slightly different though, so might not work for you.

Reply:
Yep, I was thinking the same re: having to iterate over it because it's an array. I initially decided not to reuse the `updateStep` function because the behaviour is a little bit different, but I think I'll take another look because it would make it a little cleaner.

Reply:
I've updated this to use the `updateStep` call, looks cleaner now!

Reply:
> state is an array

Doh!